bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/Observation.scala

```diff
@@ -45,60 +52,89 @@ class Observation private (val blueprint:Option[BlueprintBase],
     case None => realTarget
   }
 
+  // REL-2985 requires special handling of the times.
+  // 2017A and earlier: intTimeParam will be None, and calculatedTimesParam will be defined with progTime.
+  // If a blueprint exists - which it should - we can calculate the intTime from the progTime and then the rest
+  // of the observation times from the intTime.
+  val intTime: Option[TimeAmount] = {
+    if (intTimeParam.nonEmpty) intTimeParam
+    else for {
+      b <- blueprint
+      c <- Overheads(b)
+      p <- progTimeParam
+    } yield c.intTimeFromProgTime(p)
+  }
+
+  // 2017B and later: if the blueprint and the intTime are defined, recalculate the calculatedTimes.
+  val calculatedTimes: Option[ObservationTimes] =  for {
+    b <- blueprint
+    c <- Overheads(b)
+    t <- intTime
+  } yield c.calculate(t)
+
+  def progTime:  Option[TimeAmount] = calculatedTimes.map(_.progTime)
+  def partTime:  Option[TimeAmount] = calculatedTimes.map(_.partTime)
```

Comment:
Ok, it is kind of misleading, I thought it was partner time, which is odd as it depends on how do you apply to time to each partner